clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java

```diff
@@ -383,6 +383,16 @@ public synchronized void removeStorageEngine(String kafkaTopic) {
     factory.removeStorageEngine(storageEngine);
   }
 
+  /**
+   * This function is used to forcely clean up all the databases belonging to {@param kafkaTopic}.
+   * This function will only be used when the {@link #removeStorageEngine(String)} function can't
+   * handle some edge case, such as some partitions are lingering, which are not visible to the corresponding
+   * {@link AbstractStorageEngine}
+   */
+  public synchronized void forceStorageEngineCleanup(String kafkaTopic) {
+    persistenceTypeToStorageEngineFactoryMap.values().forEach(factory -> factory.removeStorageEngine(kafkaTopic));
```

Comment:
Do you think it is better to merge this call into removeStorageEngine directly?

Reply:
No.
For DaVinci use case, it is expected to have some leaked partitions, but for the backend use cases, that might indicate an issue.

And we can only invoke this function after the regular `removeStorageEngine` when II is enabled since both main the process and II process can host a subset of total partitions.